src/zksync-specifics/gas.md

```diff
@@ -0,0 +1,36 @@
+## Gas
+
+### Gas reported back to the EVM
+Foundry has an `isolate` mode for the EVM where all `CALL`/`CREATE` operations at the root level of a test (i.e: with depth 1) will be intercepted and treated as independent transactions. This allows for accounting of the actual transaction gas and include for example the fixed 21000 gas cost charged to the user.
+
+Running in zkVM mode is anologous to running in `isolate` mode but using the zkEVM instead. Every `CALL`/`CREATE` will be intercepted, a transaction representing the operation built, and finally a VM with that transaction in the bootloader's heap will be spawned and run in order to simulate the execution of that transaction. The gas used reported back to the EVM, and hence the one seen on traces and gas-reports, is what would be charged to the user for submitting that transaction. That value differs from the computational cost of running the called contract code and includes: 
+
+  1. Intrinsic costs: Overhead charged on each transaction.
+  2. Validation costs: Gas spent on transaction validation. May vary depending on the account making the transaction. See [Account Abstraction](https://docs.zksync.io/build/developer-reference/account-abstraction) docs.
+  3. Execution costs: Gas spent on marking factory deps and executing the transaction.
```

Comment:
From my understanding that is the `marking factory deps` part. Anyways, maybe the in depth breakdown should be in Zksync's docs and here we just give an overview so that reader understand it's not just computation?